src/eos-updater-apply.c

```diff
@@ -73,41 +79,364 @@ get_test_osname (void)
+               GError       **error)
+{
+  g_autoptr(FlatpakInstallation) installation = eos_updater_get_flatpak_installation (cancellable,
+                                                                                      error);
+  gsize i = 0;
+
+  if (!installation)
+    return FALSE;
+
+  for (; i < pending_flatpak_ref_actions->len; ++i)
+    {
+      FlatpakRemoteRefAction *action = g_ptr_array_index (pending_flatpak_ref_actions, i);
+      FlatpakRemoteRef *to_install = action->ref;
+      g_autoptr(GError) local_error = NULL;
+
+      if (action->type != FLATPAK_REMOTE_REF_ACTION_INSTALL)
+        continue;
+
+      /* We have to pass in a local_error instance here and check to see
+       * if it was FLATPAK_ERROR_ONLY_PULLED - this is what will be
+       * thrown if we succeeded at pulling the flatpak into the local
+       * repository but did not deploy it (since no FlatpakInstalledRef
+       * will be returned). */
+      flatpak_installation_install_full (installation,
+                                         FLATPAK_INSTALL_FLAGS_NO_DEPLOY,
+                                         flatpak_remote_ref_get_remote_name (to_install),
+                                         flatpak_ref_get_kind (FLATPAK_REF (to_install)),
+                                         flatpak_ref_get_name (FLATPAK_REF (to_install)),
+                                         NULL,
+                                         NULL,
+                                         NULL,
+                                         NULL,
+                                         NULL,
+                                         cancellable,
+                                         &local_error);
+
+      /* This is highly highly unlikely to happen and should usually only
+       * occurr in cases of deployment or programmer error,
+       * FLATPAK_INSTALL_FLAGS_NO_DEPLOY is documented to always return
+       * the error FLATPAK_ERROR_ONLY_PULLED */
+      if (!local_error)
+        {
+          g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED, "Flatpak installation should not have succeeded!");
+          cleanup_undeployed_flatpaks_in_installation (installation, pending_flatpak_ref_actions, i);
+          return FALSE;
+        }
+
+      /* Something unexpected failed, return early now.
+       *
+       * We are not able to meaningfully clean up here - the refs will remain
+       * in the flatpak ostree repo for the next time we want to install them
+       * but there's nothing in the public API that will allow us to get rid of
+       * them. */
+      if (!g_error_matches (local_error, FLATPAK_ERROR, FLATPAK_ERROR_ONLY_PULLED))
+        {
+          g_propagate_error (error, g_steal_pointer (&local_error));
+          cleanup_undeployed_flatpaks_in_installation (installation, pending_flatpak_ref_actions, i);
+          return FALSE;
+        }
+    }
+
+  return TRUE;
+}
+
+#if 0
+/* We return a transfer-full GFile of the record of flatpaks to be
```

Comment:
Please remove this altogether instead of commenting it out if we're not using it yet.